docs/creating_platform_code_review.md

```diff
@@ -15,63 +16,76 @@ A checklist of things to do when you're adding a new platform.
+ 5. Never depend on users adding things to `customize` to configure behavior inside your platform.
 
-    from homeassistant.const import CONF_FILENAME, CONF_HOST
-    from homeassistant.components.light import PLATFORM_SCHEMA
-    import homeassistant.helpers.config_validation as cv
+```python
+import voluptuous as vol
 
-    CONF_ALLOW_UNREACHABLE = 'allow_unreachable'
-    DEFAULT_UNREACHABLE = False
+from homeassistant.const import CONF_FILENAME, CONF_HOST
+from homeassistant.components.light import PLATFORM_SCHEMA
+import homeassistant.helpers.config_validation as cv
 
-    PLATFORM_SCHEMA = PLATFORM_SCHEMA.extend({
-        vol.Required(CONF_HOST): cv.string,
-        vol.Optional(CONF_ALLOW_UNREACHABLE,
-                    default=DEFAULT_UNREACHABLE): cv.boolean,
-        vol.Optional(CONF_FILENAME): cv.string,
-    })
-    ```
- 5. Never depend on users adding things to `customize` to configure behavior inside your platform.
+CONF_ALLOW_UNREACHABLE = 'allow_unreachable'
+DEFAULT_UNREACHABLE = False
+
+PLATFORM_SCHEMA = PLATFORM_SCHEMA.extend({
+    vol.Required(CONF_HOST): cv.string,
+    vol.Optional(CONF_ALLOW_UNREACHABLE,
+                default=DEFAULT_UNREACHABLE): cv.boolean,
+    vol.Optional(CONF_FILENAME): cv.string,
+})
+```
 
 ### 4. Setup Platform
 
- 1. Test if passed in info (user/pass/host etc.) works.
+ 1. Verify that the passed in configuration (user/pass/host etc.) works.
  2. Group your calls to `add_devices` if possible.
- 3. If platform adds extra services, format should be `<component>.<platform>_<service name>`.
+ 3. If platform adds extra services, format should be `<domain>.<service name>`. Make sure that your services [verify permissions](auth_permissions.md#checking-permissions).
 
 ### 5. Entity
 
- 1. Extend entity from component, e.g., `class HueLight(Light)`.
- 2. Avoid passing in `hass` as a parameter to the entity. When the entity has been added to Home Assistant, `hass` will be set on the entity by the helper in entity_platform.py. This means you can access `hass` as `self.hass` inside the entity.
+ 1. Extend entity from the integration you're building a platform for.
+
+    ```python
+    from homeassistant.components.light import Light
+
+    class HueLight(Light):
+        ...
+    ```
+
+ 2. Avoid passing in `hass` as a parameter to the entity. When the entity has been added to Home Assistant, `hass` will be set on the entity when the entity is added to Home Assistant. This means you can access `hass` as `self.hass` inside the entity.
  3. Do not call `update()` in constructor, use `add_entities(devices, True)` instead.
  4. Do not do any I/O inside properties. Cache values inside `update()` instead.
- 5. The state and/or attributes should not contain relative time since something happened. Instead it should store UTC timestamps.
+ 5. When dealing with time, state and/or attributes should not contain relative time since something happened. Instead it should store UTC timestamps.
```

Comment:
```suggestion
 5. When dealing with time, state and/or attributes should not contain relative time since something happened. Instead, it should store UTC timestamps.
```